### PR TITLE
MGMT-17842: Add image-based installer create config ISO

### PR DESCRIPTION
**Pull request by mresvanis**
This PR adds the image-based installer (IBI) `create config-image` and `create config-template` subcommands under the `image-based` command. 

For a detailed description of this feature please see the respective enhancement https://github.com/openshift/enhancements/blob/master/enhancements/installer/image-based-installer.md.

/cc @eranco74 @tsorya @zaneb @patrickdillon

### CODE REVIEW

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Comment by mresvanis** [comment]
@patrickdillon since the respective [enhancement](https://github.com/openshift/enhancements/pull/1617) has been merged, would you have the time to review this PR?

We would appreciate any pointers/feedback on whether we've leveraged the installer codebase as much as we should or we missed anything and of course any other comments :)

**Comment by patrickdillon** [comment]
This is targeting the master branch, and I think it should be targeting the image-based-installer feature branch.

File: pkg/asset/imagebased/configimage/installconfig.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+// InstallConfig asset.
+func (i *InstallConfig) Dependencies() []asset.Asset {
+	// Return no dependencies for the Agent install config, because it is
+	// optional. We don't need to run the survey if it doesn't exist, since the
+	// user may have supplied cluster-manifests that fully define the cluster.
+	return []asset.Asset{}
+}
+
+// Generate generates the install-config.yaml file.
+func (i *InstallConfig) Generate(parents asset.Parents) error {
```
Comment (patrickdillon) [inline pkg/asset/imagebased/configimage/installconfig.go]: heads up, since #8562 `Generate` now accepts a context

**Comment by mresvanis** [comment]
> This is targeting the master branch, and I think it should be targeting the image-based-installer feature branch.

@patrickdillon https://github.com/openshift/installer/pull/8709#issuecomment-2218131243

**Comment by zaneb** [comment]
It would be nice to see integration tests like the ones in https://github.com/openshift/installer/tree/master/cmd/openshift-install/testdata/agent for this work. Is that something you plan on?

**Comment by mresvanis** [comment]
> It would be nice to see integration tests like the ones in https://github.com/openshift/installer/tree/master/cmd/openshift-install/testdata/agent for this work. Is that something you plan on?

@zaneb definitely, I have started working on this, but it's an additional PR that's coming. ([respective](https://issues.redhat.com/browse/MGMT-18364) Jira)

File: pkg/asset/imagebased/configimage/configiso.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package configimage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
```
Comment (r4f4) [inline pkg/asset/imagebased/configimage/configiso.go:9]: This package is deprecated, you'd be nice not to use it in new code.

**Comment by r4f4** [comment]
/hold
For the deprecated errors packge which should be easy to replace.

File: pkg/asset/imagebased/configimage/configiso.go (hunk 2 above)
Comment (mresvanis) [reply pkg/asset/imagebased/configimage/configiso.go]: Thanks for catching this, I have updated this :pray:

**Comment by r4f4** [comment]
/hold cancel
Fixed.

**Pull request merged**